### PR TITLE
Add vyos_vlan DI module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

vyos_vlan module https://github.com/ansible/ansible/issues/31295
integration test

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

modules/network/vyos/vyos_vlan.py
test/integration/targets/vyos_vlan/
##### ANSIBLE VERSION

```
2.5.0
```

### CODE REVIEW

File: lib/ansible/modules/network/vyos/vyos_vlan.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: Create vlan
+  vyos_vlan:
+    vlan_id: 100
+    name: vlan-100
+    interfaces: eth1
```
Comment (kedarkekan) [inline lib/ansible/modules/network/vyos/vyos_vlan.py:63]: Does this add multiple interfaces to vlan during creation ?

File: lib/ansible/modules/network/vyos/vyos_vlan.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    state: present
+
+- name: Add interfaces to VLAN
+  vyos_vlan:
+    vlan_id: 100
+    interfaces:
```
Comment (kedarkekan) [inline lib/ansible/modules/network/vyos/vyos_vlan.py:69]: Pre-requisite is vlan is already created? If not present, does it create one and any default name assigned to the vlan?

File: lib/ansible/modules/network/vyos/vyos_vlan.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        address=dict(),
+        interfaces=dict(type='list', required=True),
+        delay=dict(default=10, type='int'),
+        state=dict(default='present',
+                   choices=['present', 'absent'])
+    )
+
+    aggregate_spec = deepcopy(element_spec)
+    aggregate_spec['vlan_id'] = dict(required=True)
+    aggregate_spec['interfaces'] = dict(required=True)
```
Comment (ganeshrn) [inline lib/ansible/modules/network/vyos/vyos_vlan.py]: Why `dict(required=True)` is added here? It is already part of the spec.

File: lib/ansible/modules/network/vyos/vyos_vlan.py (hunk 2 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/vyos/vyos_vlan.py:69]: @kedarX this is just an example to show how one can add multiple interfaces to vlan. For vyos, the vlan is created at the same time when you add interface.

File: lib/ansible/modules/network/vyos/vyos_vlan.py (hunk 1 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/vyos/vyos_vlan.py:63]: You can add either multiple interfaces or single interface.

File: lib/ansible/modules/network/vyos/vyos_vlan.py (hunk 3 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/vyos/vyos_vlan.py]: Looks like I forgot to remove this. This was added before I added the logic in element_spec. Thanks for pointing out. Please review now

File: lib/ansible/modules/network/vyos/vyos_vlan.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
```
Comment (ganeshrn) [inline lib/ansible/modules/network/vyos/vyos_vlan.py]: Please use updated boilerplate. [Reference](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/iosxr/iosxr_banner.py)

File: lib/ansible/modules/network/vyos/vyos_vlan.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+                    module.fail_json(msg='Interface {0} not configured on vlan {1}'.format(i, w['vlan_id']))
+
+
+def main():
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        vlan_id=dict(type='int', required=True),
+        name=dict(),
+        address=dict(),
```
Comment (ganeshrn) [inline lib/ansible/modules/network/vyos/vyos_vlan.py:243]: Is `address` option required as part of this module? I think address can be configured as part of `vyos_l3_interface` module.

**Review by ganeshrn — APPROVED** [review]
lgtm

**Pull request merged**